src/hypothesis/extra/typehints.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import print_function
+import sys
+from hypothesis import strategies as st
+from hypothesis.strategies import SearchStrategy
+from typing import Dict, Tuple, List, Iterator, Set, Union, Optional, NamedTuple, Callable, Sequence, Iterable, Generator
+import operator
+from collections import OrderedDict
+compose = lambda f,g: lambda *x: f(g(*x))
+# see https://docs.python.org/3/library/typing.html
+try:
```

Comment:
Function annotations are python 3 only (although you can manually set the `__annotations__` attribute if you want), but the typing library is available in python 2, as well as the [mypy](https://github.com/python/mypy) gradual typing tool. So you can still leverage NamedTuples and Unions. You just can't use `func_to_strat` without "manually" adding the annotations.